### ISSUE TITLE
Back and forth through history ?

### ISSUE THREAD

**Issue by monolux** [dimsemenov/PhotoSwipe#1459]
I have not found an option that enabled the use of the browser back and forth button to go through the shown images in a gallery. Once an image is opened in a gallery, no matter how far into the gallery one is, the back button will always close the gallery. Is this intentional? Could this feature be made an option for future releases?

I can see though how such a feature could lead to bad usability, being linked to an image in the middle of a gallery, swiping left and right a few images and now clicking/tapping the back button, some users would perhaps expect to be taken to the gallery root or website root, but instead it goes back and forth through the images they previously swiped through. This could be overcome though by explicity clicking the "close X" button top right of the opened gallery, as that is the intended use for it.

Thank you for your thoughts on this feature request.

**Comment by dimsemenov** [comment]
>  Once an image is opened in a gallery, no matter how far into the gallery one is, the back button will always close the gallery. Is this intentional?

Yes, this is intentional. For the reson that you posted in the second paragraph.

> This could be overcome though by explicity clicking the "close X" button top right of the opened gallery, as that is the intended use for it.

I don't see this as a solution, large part of users will still use back button. Typical scenario for ecomemrcei site:

- User browses catalog
- Opens product page
- Opens the product image gallery
- Browses through the gallery
- Presses back button to go back to the catalog. If this step will take tens of "back" button clicks - user will be mad.

I'll consider adding option though, thanks!
